### ISSUE TITLE
portability tests are failing with protobuf errors linux_x64_gcc4.4_bo c++

### ISSUE THREAD

**Issue by stanley-cheung** [grpc/grpc#7246]
This is blocking all PRs to be merged

https://grpc-testing.appspot.com/view/Pull%20requests/job/gRPC_portability_pull_requests/4486/language=c++,scenario=linux_x64_gcc4.4_bo/console

```
libtool: compile:  g++-4.4 -DHAVE_CONFIG_H -I. -I.. -fPIC -O0 -g -pthread -DHAVE_PTHREAD=1 -DHAVE_ZLIB=1 -Wall -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -O2 -g -DNDEBUG -MT google/protobuf/generated_message_reflection.lo -MD -MP -MF google/protobuf/.deps/generated_message_reflection.Tpo -c google/protobuf/generated_message_reflection.cc -o google/protobuf/generated_message_reflection.o
libtool: compile:  g++-4.4 -DHAVE_CONFIG_H -I. -I.. -fPIC -O0 -g -pthread -DHAVE_PTHREAD=1 -DHAVE_ZLIB=1 -Wall -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -O2 -g -DNDEBUG -MT google/protobuf/map_field.lo -MD -MP -MF google/protobuf/.deps/map_field.Tpo -c google/protobuf/map_field.cc -o google/protobuf/map_field.o
depbase=`echo google/protobuf/message.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
	/bin/bash ../libtool --tag=CXX   --mode=compile g++-4.4 -DHAVE_CONFIG_H -I. -I..   -fPIC -O0 -g  -pthread -DHAVE_PTHREAD=1 -DHAVE_ZLIB=1 -Wall -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -O2 -g -DNDEBUG -MT google/protobuf/message.lo -MD -MP -MF $depbase.Tpo -c -o google/protobuf/message.lo google/protobuf/message.cc &&\
	mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  g++-4.4 -DHAVE_CONFIG_H -I. -I.. -fPIC -O0 -g -pthread -DHAVE_PTHREAD=1 -DHAVE_ZLIB=1 -Wall -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -O2 -g -DNDEBUG -MT google/protobuf/message.lo -MD -MP -MF google/protobuf/.deps/message.Tpo -c google/protobuf/message.cc -o google/protobuf/message.o
In file included from ./google/protobuf/map_field_lite.h:34,
                 from ./google/protobuf/map_field.h:40,
                 from google/protobuf/map_field.cc:31:
./google/protobuf/map.h: In constructor 'google::protobuf::Map<Key, T>::MapAllocator<U>::MapAllocator(const google::protobuf::Map<Key, T>::MapAllocator<X>&) [with X = google::protobuf::Map<google::protobuf::MapKey, google::protobuf::MapValueRef>::KeyValuePair, U = google::protobuf::Map<google::protobuf::MapKey, google::protobuf::MapValueRef>::InnerMap::Node, Key = google::protobuf::MapKey, T = google::protobuf::MapValueRef]':
./google/protobuf/map.h:1289:   instantiated from 'U* google::protobuf::Map<Key, T>::InnerMap::Alloc(size_t) [with U = google::protobuf::Map<google::protobuf::MapKey, google::protobuf::MapValueRef>::InnerMap::Node, Key = google::protobuf::MapKey, T = google::protobuf::MapValueRef]'
./google/protobuf/map.h:958:   instantiated from 'std::pair<google::protobuf::Map<Key, T>::InnerMap::iterator_base<google::protobuf::Map<Key, T>::KeyValuePair>, bool> google::protobuf::Map<Key, T>::InnerMap::insert(const google::protobuf::Map<Key, T>::KeyValuePair&) [with Key = google::protobuf::MapKey, T = google::protobuf::MapValueRef]'
./google/protobuf/map.h:987:   instantiated from 'google::protobuf::MapPair<Key, T>*& google::protobuf::Map<Key, T>::InnerMap::operator[](const Key&) [with Key = google::protobuf::MapKey, T = google::protobuf::MapValueRef]'
./google/protobuf/map.h:1503:   instantiated from 'T& google::protobuf::Map<Key, T>::operator[](const Key&) [with Key = google::protobuf::MapKey, T = google::protobuf::MapValueRef]'
google/protobuf/map_field.cc:189:   instantiated from here
./google/protobuf/map.h:655: error: 'google::protobuf::Arena* const google::protobuf::Map<google::protobuf::MapKey, google::protobuf::MapValueRef>::MapAllocator<google::protobuf::Map<google::protobuf::MapKey, google::protobuf::MapValueRef>::KeyValuePair>::arena_' is private
./google/protobuf/map.h:590: error: within this context
make[3]: *** [google/protobuf/map_field.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory `/var/local/git/grpc/third_party/protobuf/src'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/var/local/git/grpc/third_party/protobuf'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/var/local/git/grpc/third_party/protobuf'
make: *** [/var/local/git/grpc/libs/dbg/protobuf/libprotobuf.a] Error 2
```

**Comment by stanley-cheung** [comment]
Looks like this is google/protobuf#1692

**Comment by stanley-cheung** [comment]
Looks like for the third_party entry, we need google/protobuf to be at least 	
v3.0.0-beta-3.2 commit:d4d13a4

**Comment by stanley-cheung** [comment]
Actually the before of this commit @b360c8acc had it at beta-3.2. Why did we move back to beta-3?

**Comment by stanley-cheung** [comment]
Fixed by #7247
